plugins/test/unit/plugins/distributors/test_configuration.py

```diff
@@ -145,11 +145,8 @@ def test__is_valid_repo_registry_id(self):
             'a' * 255
         ]
         should_not_be_valid = [
-            'things with spaces',
             'UPPERCASE',
             'Uppercase',
-            'upperCase',
-            'uppercasE',
```

Comment:
Could you elaborate why you decided to remove those cases. It seems like uppercase at any place should be invalid, as well as there should be no spaces... I guess I'm just missing something.

Reply:
i removed because the test was failing and i don't want to dig into this huge regexp which i took directly from docker distribution code. if there is a bug then its docker's bug.
they also have some of their tests which allow uppercase https://github.com/docker/distribution/blob/master/reference/regexp_test.go#L125

Reply:
uppercase was allowed until docker engine 1.9, now everything should be lowercase